src/zope/testrunner/tests/test_runner.py

```diff
@@ -173,10 +173,34 @@ class ZZ(K1, K2, K3): pass
         # Does that matter?  The class diagram is symmetric, so I think not.
 
     def test_FakeInputContinueGenerator_close(self):
-        # multiprocessing (and likely other forkful frameworks want to
+        # multiprocessing (and likely other forkful frameworks) want to
         # close sys.stdin.  The test runner replaces sys.stdin with a
         # FakeInputContinueGenerator for some reason. It should be
         # closeable.
 
         f = runner.FakeInputContinueGenerator()
         f.close()
+
+class TestWarnings(unittest.TestCase):
+
+    @unittest.skipIf(sys.warnoptions, "Only done if no user override")
+    def test_warning_filter_default(self):
+        # When we run tests, we run them with a 'default' simplefilter.
+        # Note that this test will fail if PYTHONWARNINGS is set,
+        # or a -W option was given, so we skip it
+        import warnings
+        # Save the current filters, ignoring the compiled regexes,
+        # which can't be compared.
+        old_filters = [(f[0], f[2], 4) for f in warnings.filters]
+        with warnings.catch_warnings():
+            # Set up just like the runner does
+            warnings.simplefilter('default')
+            warnings.filterwarnings('module',
+                                    category=DeprecationWarning,
+                                    message=r'Please use assert\w+ instead.')
+            new_filters = [(f[0], f[2], 4) for f in warnings.filters]
+
+        # For some reason, catch_warnings doesn't fully reset things,
+        # and we wind up with some duplicate entries in new_filters
+        self.assertEqual(set(old_filters), set(new_filters))
+        warnings.warn("This should be visible by default", DeprecationWarning)
```

Comment:
This is great while developing this feature, but shouldn't be left this way.  For one, it's not an automated test if it requires human verification.

Can we intercept the warning (maybe using `logging.captureWarnings(True)`) and assert that it was shown?

Reply:
good idea.